org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/ClasspathManagement.scala

```diff
@@ -493,5 +494,42 @@ trait ClasspathManagement extends HasLogger { self: ScalaProject =>
     errors.toSeq
   }
 
+  /** Make sure compiler plugins are binary compatible with the selected Scala installation.
+   *
+   *  Unlike regular dependencies, compiler plugins must be cross-compiled with the exact compiler
+   *  version used for building.
+   */
+  private def validateCompilerPlugins(): Seq[(Int, String, String)] = if (!isUsingCompatibilityMode()) Seq() else {
+    val errors = mutable.ListBuffer[(Int, String, String)]()
+
+    def paths(p: String) =
+      p.split(File.pathSeparator).toSeq.filterNot(_.isEmpty).distinct
+
+    def scanDir(d: String) = {
+      val entries = Option(new File(d).listFiles).getOrElse(Array.empty[File])
+      for (f <- entries if f.isFile)
+        yield f.getAbsolutePath
+    }
+
+    val plugins = paths(storage.getString("Xpluginsdir")).flatMap(scanDir) ++ paths(storage.getString("Xplugin"))
+    val installation = effectiveScalaInstallation()
```

Comment:
This check is there for the builder compiler, not the presentation compiler. Compiler plugins would work fine when building if they really match the Scala compiler version used for building (say, `2.10.4`).